x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/common/http/HttpClientTests.java

```diff
@@ -167,88 +167,67 @@ public void testHttps() throws Exception {
+                .put("xpack.security.transport.ssl.certificate", certPath)
                 .setSecureSettings(secureSettings)
                 .build();
 
             TestsSSLService sslService = new TestsSSLService(settings2, environment);
-            testSslMockWebserver(client, sslService.sslContext(), false);
+            testSslMockWebserver(client, sslService.sslContext("xpack.security.transport.ssl"), false);
         }
     }
 
     public void testHttpsDisableHostnameVerification() throws Exception {
         Path certPath = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode-no-subjaltname.crt");
         Path keyPath = getDataPath("/org/elasticsearch/xpack/security/transport/ssl/certs/simple/testnode-no-subjaltname.pem");
         Settings settings;
-        if (randomBoolean()) {
-            MockSecureSettings secureSettings = new MockSecureSettings();
-            Settings.Builder builder = Settings.builder()
-                .put("xpack.http.ssl.certificate_authorities", certPath);
-            if (inFipsJvm()) {
-                //Can't use TrustAllConfig in FIPS mode
-                builder.put("xpack.http.ssl.verification_mode", VerificationMode.CERTIFICATE);
-            } else {
-                builder.put("xpack.http.ssl.verification_mode", randomFrom(VerificationMode.NONE, VerificationMode.CERTIFICATE));
-            }
-            settings = builder.build();
+        Settings.Builder builder = Settings.builder()
+            .put("xpack.http.ssl.certificate_authorities", certPath);
+        if (inFipsJvm()) {
+            //Can't use TrustAllConfig in FIPS mode
+            builder.put("xpack.http.ssl.verification_mode", VerificationMode.CERTIFICATE);
         } else {
-            Settings.Builder builder = Settings.builder()
-                .put("xpack.ssl.certificate_authorities", certPath);
-            if (inFipsJvm()) {
-                //Can't use TrustAllConfig in FIPS mode
-                builder.put("xpack.ssl.verification_mode", VerificationMode.CERTIFICATE);
-            } else {
-                builder.put("xpack.ssl.verification_mode", randomFrom(VerificationMode.NONE, VerificationMode.CERTIFICATE));
-            }
-            settings = builder.build();
+            builder.put("xpack.http.ssl.verification_mode", randomFrom(VerificationMode.NONE, VerificationMode.CERTIFICATE));
         }
+        settings = builder.build();
         try (HttpClient client = new HttpClient(settings, new SSLService(settings, environment), null)) {
             MockSecureSettings secureSettings = new MockSecureSettings();
             // We can't use the client created above for the server since it only defines a truststore
-            secureSettings.setString("xpack.ssl.secure_key_passphrase", "testnode-no-subjaltname");
+            secureSettings.setString("xpack.security.transport.ssl.secure_key_passphrase", "testnode-no-subjaltname");
```

Comment:
same as above, just for being easy to the eye use the http layer settings for the MockWebServer